### DISCUSSION TITLE
How to use mappings?

### DISCUSSION THREAD

**Discussion by LuciaPauw** [mapping-commons/sssom#366] [Q&A]
To my understanding, the use of an ontological mapping lies in linking concepts from different ontologies which refer to the same or related (real-world) "thing", and SSSOM provides a neat standard for this. 

Once I have a mapping, say ON1:classA owl:equivalentClass ON2:classB, I know that all instances of classA are in fact also instances of classB thanks to this mapping.

How can I use this mapping? E.g. to run a SPARQL query to return all instances of classA (and therefore classB)? In a broad sense, I know I should use a SPARQL endpoint with a reasoner enabled. Are there any good resources for this? Any elegant way to do it? Thanks!

**Comment by matentzn** [comment]
Thanks for reaching out @LuciaPauw.

SSSOM itself does not really care how you use the mappings once they are there, but the `predicate_id` helps you to determine what you do with a mapping. Here are some examples:

1. Ontology merging. If the predicate_id is `owl:equivalentClass`, you can simply use sssom-py to transform the mapping into OWL, merge it with the ontology, and load the hole thing into a reasoner.
2. KG merging. If the predicate_id is mostly something like skos:exactMatch, and you dont want to use a reasoner, you can leverage the mapping files during KG preprocessing to merge individual graph nodes into one. The most typical approach is to (1) either create a new node in the graph and merge all equivalent nodes into it (remember to retain the original ids as node metadata or (2) pick a "clique leader" (usually a namespace, like "all anatomy terms should be merged into their respective Uberon ID...") and do the same (we do this at https://monarchinitiative.org/). Practically, we have the raw data (pre-KG) in tables, and processes that take sssom files as an input and rewire the raw data against the stuff in the sssom mapping file)
3. KG querying. Instead of merging nodes, you may simply decide to keep all nodes and the mapping links between them. We do that a lot in use cases were provenance is more important than efficiency. So you load the sssom file into your KG (either LPG or Triple store, whatever you like), and do the mapping at query time by simply adding a clause that looks for an ?entity or anything that is skos:exactMatch related to an entity. This makes queries more complex, so probably should be done with caution. Of course, since it is a KG, no one is keeping you from doing both - keep the original nodes _and_ do node merging (and query against the merged nodes in most cases, and the rest if provenance requirements arise).

There are no super easy answer to your question that fits all bills. As node merging can be a bit complicated if you have never done I would recommend the following:

1. convert your data and mapping into rdf (no reasoner)
2. load everything into a simple triplestore like rdf4j or blazegraph (I personally if you have little KG experience would instead recommend to try using neo4j desktop and load the data into neo4j instead, so you can browse it)
3. Use approach 3 above and see how far it gets you. You can always switch between them!

**Reply by LuciaPauw** [reply]
Thank you for the speedy reply! I'll give it a go.
